activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -36,8 +36,10 @@ def to_sql_and_binds(arel_or_sql_string, binds = [], preparable = nil) # :nodoc:
           else
             sql = visitor.compile(arel_or_sql_string.ast, collector)
           end
+#         sql = add_query_comment_to_sql(sql)
           [sql.freeze, binds, preparable]
         else
+#         arel_or_sql_string = add_query_comment_to_sql(arel_or_sql_string)
```

Comment:
Leaving this as an example of how we could add the comments higher in the stack.